analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/CollectionConstraintTests.cs

```diff
@@ -97,4 +97,31 @@ void Test(List<string> foo)
             }
             """)
         .Verify();
+
+    [TestMethod]
+    public void CollectionConstraint_ShouldNotLearnConstraintFromBinariesForNonTrackedTypes() =>
+        builder.AddSnippet("""
+            using System;
+            ﻿using System.Collections.Generic;
+            class NotACollection
+            {
+                List<int> items;
+                int Count => items.Count;
+            
+                void Remove(int i) => items.Remove(i);
+            
+                void Foo(NotACollection notACollection)
+                {
+                    if (notACollection.Count > 0)
+                    {
+                        notACollection.Remove(5);
+                        if (notACollection.Count > 0)   // Compliant
```

Comment:
Do we have a test where the collection is custom and derives from a tracked collection?
To cover the `.DerivesOrImplementsAny(TrackedCollectionTypes))` case.

Reply:
We have one where a custom collection is implementing `IList` but I don't think we have one for inheritance. I can add one next week.

Reply:
Sounds good!